src/backend/elfobj.c

```diff
@@ -1689,11 +1691,66 @@ STATIC void setup_comdat(Symbol *s)
     symbol_debug(s);
     if (tyfunc(s->ty()))
     {
+        reset_symbuf->write(&s, sizeof(s));
+
         //s->Sfl = FLcode;      // was FLoncecode
-        //prefix = ".gnu.linkonce.t";   // doesn't work, despite documentation
-        prefix = ".text.";              // undocumented, but works
-        type = SHT_PROGBITS;
-        flags = SHF_ALLOC|SHF_EXECINSTR;
+        /* ".gnu.linkonce.t" sections result in "Too many sections" errors from ld.
+         * prefix = ".gnu.linkonce.t";
+         */
+
+        /* Get section index of the comdat
+         */
+        IDXSEC comdatidx = section_cnt + 1;
+
+        /* create a weak symbol with the group signature in it.
+         *    4  .22 .weak _Z3barIiEiT_,@FUNCTION,VALUE=.text._Z3barIiEiT_+0x00,SIZE=14
+         */
+        const char *p = cpp_mangle(s);
+        IDXSTR namidx = Obj::addstr(symtab_strings, p);
+        IDXSYM info = elf_addsym(namidx, 0, 0, STT_FUNC, STB_WEAK, comdatidx);
```

Comment:
Just spent more than an hour on this, but I'm skeptical whether it could work out to squeeze the comdat group into setup_comdat.
It's already messy to prepone the symbol definition here (and still repeat it in Obj::comdat/func_start or wherever)
Even after making that already entangled code work somehow, it does not put other data that belongs to that function into the comdat group.
Eventually I ended up with a linker error complaining about a sth. in `.rodata` referencing a function in a discarded section.
```
`.text._D3std6format18__T10FormatSpecTaZ10FormatSpec6fillUpMFNaNfZv' referenced in section `.rodata' of generated/linux/debug/64/unittest/std/path.o: defined in discarded section `.text._D3std6format18__T10FormatSpecTaZ10FormatSpec6fillUpMFNaNfZv[_D3std6format18__T10FormatSpecTaZ10FormatSpec6fillUpMFNaNfZv]' of generated/linux/debug/64/unittest/std/path.o
```

Maybe that is a bug in my changes. But OTOH we'd also want to add EH tables into a function's COMDAT group.
How about we open that group (similar to func_start) and add the function as signature symbol after everything was emitted.
Not sure whether related symbols are all emitted with the function though.